### PR TITLE
Added SSL/TLS Modules

### PR DESCRIPTION
**Pull request by Captain-T2004**
#### Checklist
- [x] I have followed the [Contributor Guidelines](https://github.com/OWASP/Nettacker/wiki/Developers#contribution-guidelines).
- [x] The code has been thoroughly tested in my local development environment with flake8 and pylint.
- [x] The code is Python 3 compatible.
- [x] The code follows the PEP8 styling guidelines with 4 spaces indentation.
- [x] This Pull Request relates to only one issue or only one feature
- [ ] I have referenced the corresponding issue number in my commit message
- [x] I have added the relevant documentation.
- [x] My branch is up-to-date with the Upstream master branch.

#### Changes proposed in this pull request
I have added the SSL/TLS Modules to Nettacker which can detect for weak SSL versions, weak cypher suite and SSL certificate related vulnerabilities.
#### Your development environment
- OS: `Kali GNU/Linux Rolling x86_64 `
- OS Version: `6.8.11-amd64 `
- Python Version: `3.11.9`

### CODE REVIEW

File: nettacker/core/lib/ssl.py [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    for cipher in cipher_suites:
+        if test_single_cipher(host, port, cipher, timeout):
+            supported_ciphers.append(cipher)
+
+    weak_ciphers = ["LOW", "EXP", "eNULL", "aNULL", "RC4", "DES", "MD5", "DH", "ADH"]
+    for cipher in supported_ciphers:
+        if cipher in weak_ciphers:
+            return (supported_ciphers, True)
+
+    return (supported_ciphers, False)
```
Comment (arkid15r) [inline nettacker/core/lib/ssl.py:108]:
```suggestion
    weak_ciphers = {"LOW", "EXP", "eNULL", "aNULL", "RC4", "DES", "MD5", "DH", "ADH"}
    for cipher in supported_ciphers:
        if cipher in weak_ciphers:
            return supported_ciphers, True

    return supported_ciphers, False
```

**Review by arkid15r — COMMENTED** [review]
Thanks for adjusting the code to the new structure. The tests are very appreciated too!

I have a couple of minor comments and looking forward to hearing @securestep9's input

Thanks for implementing this @Captain-T2004

File: nettacker/core/lib/ssl.py [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    return False
+
+
+def is_weak_ssl_version(host, port, timeout):
+    def test_ssl_verison(host, port, timeout, ssl_version=None):
+        try:
+            context = ssl.SSLContext(ssl_version)
+            socket_connection = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            socket_connection.settimeout(timeout)
+            socket_connection.connect((host, port))
```
Comment (arkid15r) [inline nettacker/core/lib/ssl.py:29]: Could you extract it to a separate function and reuse it instead (reduce the code repetition)?

**Review by securestep9 — APPROVED** [review]
Thank you for your hard work during GSoC 2024  @Captain-T2004 !

**Pull request merged**